ui/components/app/snaps/snap-ui-renderer/snap-ui-renderer.js

```diff
@@ -42,6 +45,16 @@ const SnapUIRendererComponent = ({
     getMemoizedTargetSubjectMetadata(state, snapId),
   );
 
+  const pendingApprovals = useSelector(getUnapprovedConfirmations);
+  const history = useHistory();
+
+  useEffect(() => {
+    if (pendingApprovals.length > 0) {
```

Comment:
@FrederikBolding this change means that any pending approval will trigger this navigation. If you think this is too broad we can scope it down to specific approvals as we have done [here](https://github.com/MetaMask/metamask-extension/pull/22484/files#diff-2fbcca89731f3708d27bbd012c95c902a9049abbcdb1844b28c186a3b64d3bc8R136). I figured this behaviour may be desirable for other snaps though hence the broad strokes approach.